src/main/java/com/mongodb/hibernate/jdbc/MongoPreparedStatement.java

```diff
@@ -0,0 +1,287 @@
+/*
+ * Copyright 2024-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hibernate.jdbc;
+
+import static java.lang.String.format;
+
+import com.mongodb.client.ClientSession;
+import com.mongodb.client.MongoClient;
+import com.mongodb.hibernate.internal.NotYetImplementedException;
+import com.mongodb.hibernate.internal.VisibleForTesting;
+import java.io.InputStream;
+import java.math.BigDecimal;
+import java.sql.Array;
+import java.sql.Blob;
+import java.sql.Clob;
+import java.sql.Date;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.sql.Types;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.stream.IntStream;
+import org.bson.BsonArray;
+import org.bson.BsonBinary;
+import org.bson.BsonBoolean;
+import org.bson.BsonDateTime;
+import org.bson.BsonDecimal128;
+import org.bson.BsonDocument;
+import org.bson.BsonDouble;
+import org.bson.BsonInt32;
+import org.bson.BsonInt64;
+import org.bson.BsonNull;
+import org.bson.BsonString;
+import org.bson.BsonType;
+import org.bson.BsonValue;
+import org.bson.types.Decimal128;
+import org.jspecify.annotations.Nullable;
+
+/**
+ * MongoDB Dialect's JDBC {@link java.sql.PreparedStatement} implementation class.
+ *
+ * <p>It only focuses on API methods MongoDB Dialect will support. All the other methods are implemented by throwing
+ * exceptions in its parent {@link PreparedStatementAdapter adapter interface}.
+ */
+final class MongoPreparedStatement extends MongoStatement implements PreparedStatementAdapter {
+
+    private final BsonDocument command;
+
+    private final List<Consumer<BsonValue>> parameters;
+
+    public MongoPreparedStatement(
+            MongoClient mongoClient, ClientSession clientSession, MongoConnection mongoConnection, String mql) {
+        super(mongoClient, clientSession, mongoConnection);
+        this.command = BsonDocument.parse(mql);
+        this.parameters = new ArrayList<>();
+        parseParameters(command, parameters);
+    }
+
+    @Override
+    public ResultSet executeQuery() throws SQLException {
+        checkClosed();
+        throw new NotYetImplementedException();
+    }
+
+    @Override
+    public int executeUpdate() throws SQLException {
+        checkClosed();
+        return executeUpdate(command);
```

Comment:
In POC (and in tech doc), we implemented by invoking parent class or `MongoStatement`'s `executeUpdate(String mql)` but I think it is not the best for some reasons:

- we need to parse mql again unnecessarily; passing command directly to a new protected method accepting a parsed command makes more sense
- we had to solve the parameters in `Statement` class which should know nothing about `PreparedStatement` parameter concern; passing a command with all parameters resolved makes more sense